platforms/core-configuration/kotlin-dsl/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/TestKitIntegrationTest.kt

```diff
@@ -250,6 +252,12 @@ class TestKitIntegrationTest : AbstractKotlinIntegrationTest() {
             """
         )
 
+        // Kotlin compiler uses sun.misc.Unsafe that prints a warning with Java 24
+        Jvm.current().javaVersion?.let {
```

Comment:
Hmmm... aren't JDK warning checks disabled by default? `AbstractGradleExecutor.jdkWarningChecksOn`

Reply:
That is a good point. So this fix is not good. I will replace it with:
https://github.com/gradle/gradle/pull/33241